taichi/analysis/bls_analyzer.cpp

```diff
@@ -44,14 +44,17 @@ void BLSAnalyzer::record_access(Stmt *stmt, AccessFlag flag) {
     }
     bool matching_indices = true;
     std::vector<IndexRange> offsets;
+    std::vector<int> coeffs;
     offsets.resize(ptr->indices.size());
+    coeffs.resize(ptr->indices.size());
     const int num_indices = (int)ptr->indices.size();
     for (int i = 0; i < num_indices; i++) {
       auto diff = irpass::analysis::value_diff_loop_index(ptr->indices[i],
                                                           for_stmt_, i);
-      if (diff.linear_related()) {
+      if (diff.related_() and diff.coeff > 0) {
```

Comment:
s/and/&&, msvc doesn't seem to support `and` 😂

Reply:
Oops! I had no idea that was even valid syntax with any compiler! will fix